userspace/libsinsp/filter.cpp

```diff
@@ -1294,9 +1308,294 @@ void sinsp_filter::pop_expression()
 	m_nest_level--;
 }
 
+//
+// Check to see if the filter has 1 or more  SQL 'in' clause(s)
+// return modified filter to "or" clauses, else return the original filter string
+// throw if no filter field before SQL "in" delimitor, or no "," after multiple values
+//
+string sinsp_filter::parse_sql_in_clause(const string& fltstr)
+{
+	// Will contain each filter component splitting on " "
+	vector<string> components;
+
+	// The return fltstr which has been modified
+	string modified_fltstr;
+
+	// Tmp storage of each operand 'filter' e.g. proc.name
+	string operand;
+
+	// Flags to keep the state while transforming the SQL 'in' clause to the sysdig 'or' syntax
+	bool in_state = false;
+	bool start_in_state = false;
+	bool done_in_state = false;
+	bool value_state = false;
+	bool start_value_state = false;
+	bool done_value_state = false;
+
+	// Count to make sure "," exists within SQL "in" clause
+	uint32_t value_count = 0;
+	uint32_t comma_count = 0;
+
+	// Only parse the filter string if it has SQL "in" clause(s)
+	if(strcasestr(fltstr.c_str(), " in") != NULL || strcasestr(fltstr.c_str(), " in(") != NULL)
```

Comment:
I was preparing for case: "IN" or "in", and it was working with the original implementation. Not that I switch to string == it isn't working and this can be changed.